pyfifotax/report_data.py

```diff
@@ -0,0 +1,548 @@
+                self.held_forex[event.currency].push(event.received_net_dividend)
+
+                if event.withheld_tax is not None:
+                    self.misc["Tax Withholding"].append(event.withheld_tax)
+                self.misc["Dividend Payments"].append(event.received_dividend)
+
+                # dividends should be small enough to not trigger AWV reportings
+
+            elif isinstance(event, BuyEvent):
+                # if not enough money, pop on FOREX Queue will fail
+                tmp = self.held_forex[event.currency].pop(
+                    event.cost_of_shares,
+                    to_decimal(1),
+                    event.date,
+                )
+                self.sold_forex[event.currency].extend(tmp)
+
+                self.held_shares[event.symbol].push(event.received_shares)
+                if event.paid_fees is not None:
+                    self.misc["Fees"].append(event.paid_fees)
+
+                self.awv_z10_events.append(
+                    awv.AWVEntryZ10Buy(
+                        date=event.date,
+                        symbol=event.received_shares.symbol,
+                        currency=event.currency,
+                        quantity=event.received_shares.quantity,
+                        value=event.received_shares.total_buy_value(),
+                    )
+                )
+
+            elif isinstance(event, SellEvent):
+                # if not enough shares to sell, pop on SHARE Queue will fail
+                tmp = self.held_shares[event.symbol].pop(
+                    event.quantity, event.sell_price, event.date
+                )
+                self.sold_shares[event.symbol].extend(tmp)
+
+                self.held_forex[event.currency].push(event.received_forex)
+
+                if event.paid_fees is not None:
+                    self.misc["Fees"].append(event.paid_fees)
+
+                self.awv_z10_events.append(
+                    awv.AWVEntryZ10Sale(
+                        date=event.date,
+                        symbol=event.symbol,
+                        currency=event.currency,
+                        quantity=event.quantity,
+                        value=event.quantity * event.sell_price,
+                    )
+                )
+
+            elif isinstance(event, MoneyDepositEvent):
+                new_forex = FIFOForex(
+                    currency=event.currency,
+                    quantity=(
+                        event.amount - event.fees.amount
+                        if event.fees is not None
+                        else event.amount
+                    ),
+                    buy_date=event.buy_date,
+                    source=f"Deposit of Money",
+                )
+                self.held_forex[event.currency].push(new_forex)
+                if event.fees is not None:
+                    self.misc["Fees"].append(event.fees)
+
+            elif isinstance(event, MoneyWithdrawalEvent):
+                try:
```

Comment:
If I try to withdraw "0 EUR" for "2.5 EUR fee" (this is how I model various non-stock related fees, e.g. exchange fees), I get this in the logs:

```
[INFO] Withdrew money on 2019-01-31: 0.00 EUR for 2.50 EUR additional fee
[DEBUG] [Σ 339.26 EUR on 2019-01-31] Popping 2.50 EUR from the queue
[DEBUG] [Σ 341.76 EUR on 2019-01-31] Popping -2.50 EUR from the queue
```

It seems the fee gets popped twice, but once with a negative sign.

Reply:
This is one of the things I am not too happy with at the moment: whether amounts include fees or not.
Technically, the money transfer values are the gross amounts, not the net amounts. (a typical use-case for me would be to issue a withdrawal of 1000 USD on Schwab and then have the 15 USD deducted without me having to do the mental math of already deducting the fee beforehand). 
This implicit assumption, however, then doesn't work for 0 EUR deposits. Should we add a separate tab for fees in general?

Reply:
Exactly my problem with IBKR FX currently: it always deducts the fees in the base currency (probably EUR in most of the cases) from the account itself. So I can't write FX fees neither to "source_fees", nor to "target_fees" because both of them reduces the converted amount.

A separate fee tab would be a good solution for this problem, yes. If someone, who fills this spreadsheet manually, wants to deal with fees, it's possible, but it's also very easy to skip it if they decide they don't want to bother.

When using one of a converter scripts, of course, it should be filled and complete.

Reply:
I will add a special case though that if the "deposit" amount is 0, just the fees a popped.